crates/types/src/event.rs

```diff
@@ -173,6 +173,70 @@ pub enum EventType<TYPES: NodeType> {
     /// A message destined for external listeners was received
     ExternalMessageReceived(Vec<u8>),
 }
+
+// Implement Eq as well, since EventType should have reflexive equality
+impl<TYPES: NodeType> Eq for EventType<TYPES> {}
+
+impl<TYPES: NodeType> PartialEq for EventType<TYPES> {
```

Comment:
This might be a bit overkill if the above external event changes are removed. If we cannot derive Eq and PartialEq here, perhaps we should consider fixing the types within EventType to do so instead, that way we can avoid arduously maintaining this for a yet-unneeded change.

Reply:
+1 this should be a `#derive`. I think you might need to derive `Eq` and/or `PartialEq` for a few types, but I don't think that should be an issue

Reply:
@jparr721, @ss-es, I agree that it’s not an ideal solution. My initial approach was to add `#[derive(Eq, PartialEq)]` annotations to the dependent inner types. However, this would have required modifying a significant number of files, so I considered handling it in a separate PR to reduce the review effort.

Please let me know if you prefer that I include these changes in this PR or if it would be better to address them in a separate PR.

Reply:
I think I'm just strongly opposed to a custom `Eq` or `PartialEq` implementation.

But after looking over the PR some more, I'm not sure if any of this is really that necessary; I think I agree with @jparr721 that the external event checking could just be removed altogether.

We might have to discuss internally about whether we want to test this (and to what extent), but I think I would just split this off from the PR